imagepipeline-backends/imagepipeline-okhttp/src/main/java/com/facebook/imagepipeline/backends/okhttp/OkHttpNetworkFetchProducer.java

```diff
@@ -83,7 +85,17 @@ protected void fetchImage(final NfpRequestState requestState) {
           new BaseProducerContextCallbacks() {
             @Override
             public void onCancellationRequested() {
-              call.cancel();
+              // Avoid NetowkrOnMainThreadException
+              if (Looper.myLooper() != Looper.getMainLooper()) {
+                call.cancel();
+              } else {
+                new AsyncTask<Call, Void, Void>() {
```

Comment:
`AsyncTask` is overkill here.  Specifically, `AsyncTask` is needed only when you require a response in some fashion from the background job on a Handler-based thread (typically the main thread).  In this case, you have fire-and-forget semantics, which a simple Thread or Executor will do much better.

I'm not as familiar with Fresco's code, but I suspect there is a standard internal Executor (or something just like it) that can be used for this purpose as well.